react/src/views/Box/BoxView.test.tsx

```diff
@@ -0,0 +1,416 @@
+        await waitFor(waitTillLoadingIsDone);
+        const boxSubheading = screen.getByTestId("box-subheader");
+        expect(boxSubheading).toHaveTextContent("State: Lost");
+    });
+    // Test case 3.1.1.2.1
+    it("3.1.1.2.1 - change box state color respectfully", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        let color;
+        const currentState = mocks[0].result.data.box?.state;
+        if (currentState === "Lost" || currentState === "Scrap") {
+            color = "#EB404A";
+        } else {
+            color = "#0CA789";
+        }
+        expect(screen.getByTestId("box-state")).toHaveStyle(`color: ${color}`);
+    });
+    // Test case 3.1.1.3
+    it("3.1.1.3 - click on + and - to increase or decrease number of items", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        let numberOfItemWhenIncreased = 32;
+        // let numberOfItemWhenDecreased = 31;
+        fireEvent.click(screen.getByTestId("increase-items"));
+        await waitFor(() => userEvent.type(screen.getByTestId("increase-number-items"), "1"));
+        fireEvent.click(screen.getByText("Submit"));
+
+        await waitFor(() => {
+            expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+            expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+                "# " + numberOfItemWhenIncreased,
+            );
+        });
+
+        // fireEvent.click(screen.getByTestId("decrease-items"));
+        // await waitFor(() => userEvent.type(screen.getByTestId("decrease-number-items"), "1"));
+
+        // await waitFor(() => {
+        //     expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+        //     expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+        //         "# " + numberOfItemWhenDecreased,
+        //     );
+        // });
+    });
+    // Test case 3.1.1.4
+    it("3.1.1.4 - clicking on Lost / Scrap must change box state respectfully", async () => {
+        await waitFor(waitTillLoadingIsDone);
+
+        fireEvent.click(screen.getByTestId("box-lost-btn"));
+        await waitFor(() => {
+            const boxSubheading = screen.getByTestId("box-subheader");
+            expect(boxSubheading).toHaveTextContent(`State: Lost`);
+            expect(screen.getByTestId("box-state")).toHaveStyle(`color: #EB404A`);
+        });
+
+        fireEvent.click(screen.getByTestId("box-scrap-btn"));
+        await waitFor(() => {
+            const boxSubheading = screen.getByTestId("box-subheader");
+            expect(boxSubheading).toHaveTextContent(`State: Scrap`);
+            expect(screen.getByTestId("box-state")).toHaveStyle(`color: #EB404A`);
+        });
+    });
+    // Test case 3.1.1.5
+    it("3.1.1.5 - click on a location should move box to seleted place", async () => {
+        await waitFor(waitTillLoadingIsDone);
+
+        // fireEvent.click(screen.getByTestId("location-wh1-btn"));
```

Comment:
still to be implemented?

Reply:
@pylipp 

> I could understand and follow most tests :) do you intend to implement the other tests from the 3.1.1 section in this PR or in another one?

@pylipp I think it make sense to skip other tests for another PR and also as we discussed we just implement few important one :)